### PR TITLE
Linked Posts ORM work for Venues

### PR DESCRIPTION
**Pull request by sc0ttkclark**
https://central.tri.be/issues/116277

### CODE REVIEW

File: src/Tribe/Repositories/Venue.php [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+		) );
+	}
+
+	/**
+	 * Filter by LIKE or REGEX string.
+	 *
+	 * @param string $meta_key Meta key to filter by.
+	 * @param string $value    MySQL compatible LIKE or REGEX string.
+	 */
+	public function filter_by_like_regex( $meta_key, $value ) {
```
Comment (lucatume) [inline src/Tribe/Repositories/Venue.php]: The same observation done for Organizers applies here: move it to the base `Tribe__Repository` class.

File: src/Tribe/Repositories/Venue.php [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+		$this->by( 'meta_like', $meta_key, $value );
+	}
+
+	/**
+	 * Filters venues by a specific address line.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $value MySQL compatible LIKE or REGEX string.
```
Comment (lucatume) [inline src/Tribe/Repositories/Venue.php]: Here and in any other place where it appears: MySQL uses `REGEXP` with a final `P`. 
If we note it to refer to SQL use `REGEXP`, not `REGEX`.

File: src/Tribe/Repositories/Venue.php [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+			'state_province' => array( $this, 'filter_by_state_province' ),
+			'website'        => array( $this, 'filter_by_website' ),
+		) );
+	}
+
+	/**
+	 * Filter by LIKE or REGEX string.
+	 *
+	 * @param string $meta_key Meta key to filter by.
+	 * @param string $value    MySQL compatible LIKE or REGEX string.
```
Comment (lucatume) [inline src/Tribe/Repositories/Venue.php]: Here and in any other place where it appears: MySQL uses REGEXP with a final P.
If we note it to refer to SQL use REGEXP, not REGEX.

**Comment by lucatume** [comment]
Same backport work that should be done in organizers should be done here and fix the tests.

File: src/Tribe/Repositories/Venue.php (hunk 1 above)
Comment (sc0ttkclark) [reply src/Tribe/Repositories/Venue.php]: I told Matt this is what I was going to do, glad we're on the same page :)

File: src/Tribe/Repositories/Venue.php (hunk 2 above)
Comment (sc0ttkclark) [reply src/Tribe/Repositories/Venue.php]: Copy/pasted from the spec doc, will update

File: src/Tribe/Repositories/Venue.php (hunk 1 above)
Comment (sc0ttkclark) [reply src/Tribe/Repositories/Venue.php]: Copy/pasted from the spec doc, will update

File: src/Tribe/Repositories/Venue.php (hunk 3 above)
Comment (sc0ttkclark) [reply src/Tribe/Repositories/Venue.php]: Copy/pasted from the spec doc, will update

**Comment by sc0ttkclark** [comment]
Moved into #2222

**Pull request closed without merging**